include/eve/detail/function/simd/x86/slice.hpp

```diff
@@ -91,15 +95,25 @@ namespace eve::detail
   //================================================================================================
   // Both slice
   //================================================================================================
-  template<typename T, typename N>
+  template<arithmetic_scalar_value T, typename N>
   EVE_FORCEINLINE auto slice(wide<T, N> const &a) noexcept
       requires x86_abi<abi_t<T, N>>
   {
-    std::array<wide<T, typename N::split_type>, 2> that{slice(a, lower_), slice(a, upper_)};
-    return that;
+    if constexpr (has_plain_translation<T>)
+    {
+      using sub_t = as<as_wide_t<T, typename N::split_type>>;
+
+      auto [l, h] = bit_cast(a, as<wide<translate_t<T>, N>>{}).slice();
+      return std::array{bit_cast(l, sub_t{}), bit_cast(h, sub_t{})};
+    }
```

Comment:
No - in x86 you definetly shouldn't do this. In generic code maybe. You probably can be helped with some better concepts.

Did you consider using `plain_scalar_value`